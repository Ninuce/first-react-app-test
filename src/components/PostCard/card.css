.card {
  /* max-width: 400px; */
  display: flex;
  flex-direction: column;
  padding: 20px 24px;
  background-color: #fff;
  margin-top: 20px;
  margin-bottom: 20px;
  border-radius: 8px;
}

.card__header {
  display: flex;
  justify-content: flex-start;
  padding-top: 20px;
  padding-bottom: 12px;
}

.card__title {
  text-align: left;
  font-size: 20px;
  margin: 0;
  padding-left: 14px;
}
.card__avatar {
  height: 34px;
  width: 34px;
  border-radius: 50%;

}
.card__media {
  width: 400px;
}

.card__actions {
  display: flex;
  justify-content: space-between;
  padding-top: 14px;
}
.card__icon {
  height: 30px;
}

.card__icon:hover {
  opacity: 0.3;
}

.card__icon:not(:last-child) {
  padding-right: 10px;
}
