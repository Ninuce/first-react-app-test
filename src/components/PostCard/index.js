import React from 'react';

import me from '../../images/me.JPG';
import heart from '../../images/heart.png';
import comment from '../../images/comment.png';
import bookmark from '../../images/bookmark.svg';
import avatar from '../../images/avatar.JPG';
import './card.css';

export const sum= (a,b) => a + b



const PostCard = ({user, text, img}) => {
  return (
  <div className="card">
    <div className="card__header">
      <img src={avatar} className="card__avatar" alt="avatar" />
      <h2 className="card__title">{user}</h2>
    </div>
    <img src={img} className="card__media" alt="logo" />

    <div className="card__actions">
      <div className="card__actions-interaction">
        <img src={heart} className="card__icon" alt="heart" />
        <img src={comment} className="card__icon" alt="comment" />
      </div>

      <div className="card__actions-personal">
        <img src={bookmark} className="card__icon" alt="bookmark" />
      </div>
    </div>
    <p>{text}</p>
  </div>
)}


export default PostCard;
