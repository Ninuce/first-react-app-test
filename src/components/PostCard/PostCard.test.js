import React from "react"
import { shallow } from "enzyme";
import toJson from "enzyme-to-json"
import PostCard, {sum} from "./index"

describe("testing postcard", () => {
  const wrapper = shallow(<PostCard/>)

  it('sums 2 numbers', () => {
    expect(sum(3, 5)).toEqual(8);
  });

  it('renders properly', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('renders snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

})
