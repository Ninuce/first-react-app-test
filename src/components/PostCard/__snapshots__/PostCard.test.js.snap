// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing postcard renders snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card__header"
  >
    <img
      alt="avatar"
      className="card__avatar"
      src="avatar.JPG"
    />
    <h2
      className="card__title"
    />
  </div>
  <img
    alt="logo"
    className="card__media"
  />
  <div
    className="card__actions"
  >
    <div
      className="card__actions-interaction"
    >
      <img
        alt="heart"
        className="card__icon"
        src="heart.png"
      />
      <img
        alt="comment"
        className="card__icon"
        src="comment.png"
      />
    </div>
    <div
      className="card__actions-personal"
    >
      <img
        alt="bookmark"
        className="card__icon"
        src="bookmark.svg"
      />
    </div>
  </div>
  <p />
</div>
`;
